Indicators/HilbertTransformIndicator.cs

```diff
@@ -0,0 +1,162 @@
+            _ => Quad!.IsReady ? _v2 - _v1 * qMult + _quad2.Peek().Value * qMult : decimal.Zero,
+            _ => _v1.IsReady && _v2.IsReady,
+            () =>
+            {
+                _v1.Reset();
+                _v2.Reset();
+                _quad2.Clear();
+            });
+    }
+
+    /// <summary>
+    /// Creates a new Hilbert Transform indicator with default name and default params
+    /// </summary>
+    /// <param name="length">The length of the FIR filter used in the calculation of the Hilbert Transform.
+    /// This parameter determines the number of filter coefficients in the FIR filter.</param>
+    /// <param name="iMult">The multiplication factor used in the calculation of the in-phase component
+    /// of the Hilbert Transform. This parameter adjusts the sensitivity and responsiveness of
+    /// the transform to changes in the input signal.</param>
+    /// <param name="qMult">The multiplication factor used in the calculation of the quadrature component of
+    /// the Hilbert Transform. This parameter also adjusts the sensitivity and responsiveness of the
+    /// transform to changes in the input signal.</param>
+    public HilbertTransformIndicator(int length = 7, decimal iMult = 0.635M, decimal qMult = 0.338M)
+        : this($"Hilbert({length}, {iMult}, {qMult})", length, iMult, qMult)
+    {
+    }
+
+    /// <summary>
+    /// Gets a flag indicating when this indicator is ready and fully initialized
+    /// </summary>
+    public override bool IsReady => updatesCount >= WarmUpPeriod;
+
+    /// <summary>
+    /// Computes the next value of this indicator from the given state
+    /// </summary>
+    /// <param name="input">The input given to the indicator</param>
+    /// <returns>A new value for this indicator</returns>
+    protected override decimal ComputeNextValue(IndicatorDataPoint input)
+    {
+        Debug.Assert(input != null, nameof(input) + " != null");
+
+        updatesCount += 1;
+
+        _input.Update(input);
+        _prev.Update(input);
+        _v1.Update(input);
+        _v2.Update(_v1.Current);
+        _v4.Update(_v1.Current);
+
+        InPhase.Update(input);
+        _inPhase3.Enqueue(InPhase.Current);
+        while (_inPhase3.Count > InPhase3Length)
+        {
+            _inPhase3.Dequeue();
+        }
+
+        Quad.Update(input);
+        _quad2.Enqueue(Quad.Current);
+        while (_quad2.Count > Quad2Length)
+        {
+            _quad2.Dequeue();
+        }
+
+        return input.Value;
+    }
+
+    /// <summary>
+    /// Required period, in data points, for the indicator to be ready and fully initialized.
+    /// </summary>
+    public int WarmUpPeriod => Math.Max(_inPhase3WarmUpPeriod, _quad2WarmUpPeriod);
+
+    /// <summary>
+    /// Resets this indicator to its initial state
+    /// </summary>
+    public override void Reset()
+    {
+        base.Reset();
+        InPhase.Reset();
+        Quad.Reset();
+        updatesCount = 0;
```

Comment:
All the indicators `_prev`, `input`, `_v1` , `_v2` ,  and `_v4`  should call Reset.

Reply:
fixes b005a02c353facb021e864c5780267125222dbc1